source/npm/util.js

```diff
@@ -107,3 +109,74 @@ exports.checkIgnoreStrategy = ({files}) => {
 		`);
 	}
 };
+
+// New files added part of .npmignore or not part of files-attribute in "package.json"
+exports.checkNewFiles = async (newFiles, filesFromPackageDotJson) => {
+	const rootDir = pkgDir.sync();
+	const npmignoreExists = fs.existsSync(path.resolve(rootDir, '.npmignore'));
+
+	let result = [];
+
+	if (filesFromPackageDotJson) {
+		result = getFilesNotPartOfFileAttribute(filesFromPackageDotJson, newFiles);
+	}
+
+	if (npmignoreExists) {
+		result = result.concat(await getFilesIgnoredByDotnpmignore(newFiles));
+	}
+
+	return result;
+};
+
+async function getFilesIgnoredByDotnpmignore(fileList) {
+	if (!Array.isArray(fileList)) {
+		throw new TypeError('expected array, but got {typeof fileList}');
+	}
+
+	const result = [];
+	await ignoreWalker({
+		path: pkgDir.sync(),
+		ignoreFiles: ['.npmignore']
+	})
+		.then(WhiteList => {
+			if (!Array.isArray(WhiteList)) {
+				throw new TypeError(`Expected array, but got ${typeof WhiteList} `);
+			}
+
+			// Every file in fileList should be part of the WhiteList
+			fileList.forEach(item => {
+				const found = WhiteList.find(WhiteListItem => {
+					return WhiteListItem === item;
+				});
+				if (found === undefined) {
+					result.push(item);
+				}
+			});
+		});
+
+	return result;
+}
+
+function getFilesNotPartOfFileAttribute(filesFromPackageDotJson, fileList) {
+	if (!Array.isArray(fileList)) {
+		throw new TypeError(`expected array, but got ${typeof fileList}`);
+	}
+
+	const result = [];
+	const globMatcher = new Minimatch(getGlobPattern(filesFromPackageDotJson),
```

Comment:
I'm not sure, but I think that 

```js
getGlobPattern(filesFromPackageDotJson)
```

can be replaced by:

```js
'{' + filesFromPackageDotJson.join(',') + '}';
```

I think it works regardless of `filesFromPackageDotJson.length`

Reply:
The ```getGlobPattern```-function is still necessary. 
If you try minimatch-filtering with a single glob-pattern in curly brackets:
```
var fileList = ['foo', 'bar'];
console.log(fileList.filter(minimatch.filter({foo}, {matchBase: true})));
```
it prints an empty array.
A single glob-pattern shouldn't be enclosed in curly brackets.
The curly brackets can only be used for multiple glob-patterns.
You could try it out with the unit-tests in file ```test/npmignore.js```.